lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -47,11 +48,39 @@ sub _git_clone_all ($job, $clones) {
 
     # iterate clones sorted by path length to ensure that a needle dir is always cloned after the corresponding casedir
     for my $path (sort { length($a) <=> length($b) || $a cmp $b } keys %$clones) {
-        my $url = $clones->{$path};
         die "Don't even think about putting '..' into '$path'." if $path =~ /\.\./;
+
+        my $git = OpenQA::Git->new(app => $app, dir => $path);
+        my $origin_url = -d $path ? $git->get_origin_url : undef;
+        my $url = $origin_url // $clones->{$path};
+        my $server_host;
+
+        if ($url =~ m{^[^@]+@([^:]+):}) {
+            # e.g. "git@gitlab.suse.de:qa/foo", "user@host:repo"
+            $server_host = $1;
+        }
+        else {
+            my $mojo_url = Mojo::URL->new($url);
+            $server_host = $mojo_url->host // 'other';
+        }
```

Comment:
This code is not fully covered and should probably be moved into its own function, e.g. `_server_host_from_git_remote`.